File: 3033-70-1014-best-sightseeing-pair/3033-70-1014-best-sightseeing-pair.js
var maxScoreSightseeingPair = function(values) {
    const n = values.length;
    let mx = 0;
    let dp = values[0];
    for (let j = 1; j < n; j++) {
        mx = Math.max(mx, values[j] - j + dp);
        dp = Math.max(dp, values[j] + j);
    }
    return mx;
};

File: 3033-70-1014-best-sightseeing-pair/3033-70-1014-best-sightseeing-pair.java
class Solution {
    public int maxScoreSightseeingPair(int[] values) {
        int n = values.length;
        int mx = 0;
        int dp = values[0];
        for (int j = 1; j < n; j++) {
            mx = Math.max(mx, values[j] - j + dp);
            dp = Math.max(dp, values[j] + j);
        }
        return mx;
    }
}
